From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 9 Apr 2020 09:02:41 +0200
Subject: [PATCH] {BUG in 5.7-rc1} drivers/net/ethernet/3com: include
 vermagic.h after module.h

gcc on x86-32 reports:

    In file included from include/linux/module.h:30,
                     from drivers/net/ethernet/3com/3c515.c:56:
    arch/x86/include/asm/module.h:73: error: "MODULE_ARCH_VERMAGIC"
    redefined [-Werror]
       73 | # define MODULE_ARCH_VERMAGIC MODULE_PROC_FAMILY
          |
    In file included from drivers/net/ethernet/3com/3c515.c:25:
    include/linux/vermagic.h:28: note: this is the location of the
    previous definition
       28 | #define MODULE_ARCH_VERMAGIC ""
          |

Because module.h defines MODULE_ARCH_VERMAGIC if CONFIG_X86_32.
---
 drivers/net/ethernet/3com/3c515.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/3com/3c515.c b/drivers/net/ethernet/3com/3c515.c
index 90312fcd6319..acdc94608693 100644
--- a/drivers/net/ethernet/3com/3c515.c
+++ b/drivers/net/ethernet/3com/3c515.c
@@ -22,7 +22,6 @@
 
 */
 
-#include <linux/vermagic.h>
 #define DRV_NAME		"3c515"
 
 #define CORKSCREW 1
@@ -54,6 +53,7 @@ static int max_interrupt_work = 20;
 #define PKT_BUF_SZ		1536	/* Size of each temporary Rx buffer. */
 
 #include <linux/module.h>
+#include <linux/vermagic.h>
 #include <linux/isapnp.h>
 #include <linux/kernel.h>
 #include <linux/netdevice.h>
-- 
